### ISSUE TITLE
Flutter run is not working

### ISSUE THREAD

**Issue by x8Haider** [flutter/flutter#41131]
Hello,

I have below issue while performing flutter run:
```
Gradle task assembleDebug failed with exit code 1

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      _buildGradleProjectV2 (package:flutter_tools/src/android/gradle.dart:751:5)
<asynchronous suspension>
#2      buildGradleProject (package:flutter_tools/src/android/gradle.dart:494:14)
<asynchronous suspension>
#3      buildApk (package:flutter_tools/src/android/apk.dart:34:9)
<asynchronous suspension>
#4      AndroidDevice.startApp (package:flutter_tools/src/android/android_device.dart:486:13)
<asynchronous suspension>
#5      FlutterDevice.runHot (package:flutter_tools/src/resident_runner.dart:378:54)
<asynchronous suspension>
#6      HotRunner.run (package:flutter_tools/src/run_hot.dart:254:39)
<asynchronous suspension>
#7      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:472:37)
<asynchronous suspension>
#8      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:490:18)
#9      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#10     _rootRunUnary (dart:async/zone.dart:1132:38)
#11     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#12     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#13     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#14     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#15     Future._completeWithValue (dart:async/future_impl.dart:522:5)
#16     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30:15)
#17     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288:13)
#18     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
#19     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#20     _rootRunUnary (dart:async/zone.dart:1132:38)
#21     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#22     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#23     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#24     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#25     Future._completeWithValue (dart:async/future_impl.dart:522:5)
#26     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30:15)
#27     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288:13)
#28     AndroidDevice.isLocalEmulator (package:flutter_tools/src/android/android_device.dart)
#29     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#30     _rootRunUnary (dart:async/zone.dart:1132:38)
#31     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#32     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#33     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#34     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#35     Future._completeWithValue (dart:async/future_impl.dart:522:5)
#36     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:552:7)
#37     _rootRun (dart:async/zone.dart:1124:13)
#38     _CustomZone.run (dart:async/zone.dart:1021:19)
#39     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#40     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#41     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#42     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#43     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116:13)
#44     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173:5)



flutter doctor -v
[√] Flutter (Channel stable, v1.9.1+hotfix.2, on Microsoft Windows [Version 10.0.18362.356], locale en-US)
    • Flutter version 1.9.1+hotfix.2 at G:\flutter
    • Framework revision 2d2a1ffec9 (2 weeks ago), 2019-09-06 18:39:49 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at G:\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = G:\Android\sdk
    • Java binary at: G:\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at G:\Android\Android Studio
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!
```

**Comment by Harshdeep4295** [comment]
Do one thing, just open the android folder only in android studio, and check whether the Gradle is synced or not. In my case that was the error, Gradle was not synced.

**Comment by escamoteur** [comment]
Please add the output of flutter doctor here.
Since when does this happen?
Did you switch channel or upgraded Flutter?

**Comment by galstc05** [comment]
The problem for me was that in some of the manifest.xml files there was `meta-unit.data` instead of `meta-data`. After I upgraded `firebase_core`, `firebase_analytics` and `firebase_crashlytics` and manually changed the manifest for `esys_flutter_share` builds are working again. Not quite sure what's causing the problem though...

**Comment by zanderso** [comment]
@blasten Do we print more information in the logs when gradle fails like this?

**Comment by blasten** [comment]
We don't on stable. However, I started dumping the stdout and stderr before we throw the `ToolExit` when we build the plugins in the tool, but it's worth adding that logic to this call since it's the most common call to Gradle.

**Comment by x8Haider** [comment]
@escamoteur 
> Please add the output of flutter doctor here.
> Since when does this happen?
> Did you switch channel or upgraded Flutter?

flutter doctor -v
[√] Flutter (Channel stable, v1.9.1+hotfix.2, on Microsoft Windows [Version 10.0.18362.356], locale en-US)
    • Flutter version 1.9.1+hotfix.2 at G:\flutter
    • Framework revision 2d2a1ffec9 (2 weeks ago), 2019-09-06 18:39:49 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at G:\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = G:\Android\sdk
    • Java binary at: G:\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at G:\Android\Android Studio
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!


The issue started yesterday only. The build was fine & have not changed things on the codes except I had upgraded to the latest flutter update before 3 days from now.

**Comment by x8Haider** [comment]
@Harshdeep4295 

I'll check this and let you know. 

So desperate as I have a meeting for this and can't give timing for it.

**Comment by x8Haider** [comment]
@Harshdeep4295 

I got this on "Run build":

In project 'app' a resolved Google Play services library dependency depends on another at an exact version (e.g. "[11.2.
0]", but isn't being resolved to that version. Behavior exhibited by the library will be unknown.

Dependency failing: com.google.android.gms:play-services:11.2.0 -> com.google.firebase:firebase-common@[11.2.0], but fir
ebase-common version was 19.0.0.

The following dependencies are project dependencies that are direct or have transitive dependencies that lead to the art
ifact with the issue.
-- Project 'app' depends on project 'cloud_firestore' which depends onto com.google.firebase:firebase-firestore@19.0.0
-- Project 'app' depends onto com.google.firebase:firebase-firestore@{strictly 19.0.0}
-- Project 'app' depends onto com.google.firebase:firebase-dynamic-links@{strictly 11.2.0}
-- Project 'app' depends onto com.google.firebase:firebase-database@{strictly 11.2.0}
-- Project 'app' depends on project 'firebase_core' which depends onto com.google.firebase:firebase-core@16.0.9
-- Project 'app' depends onto com.google.android.gms:play-services@11.2.0
-- Project 'app' depends onto com.google.firebase:firebase-storage@{strictly 11.2.0}
-- Project 'app' depends onto com.google.firebase:firebase-config@{strictly 11.2.0}
-- Project 'app' depends onto com.google.firebase:firebase-appindexing@{strictly 11.2.0}
-- Project 'app' depends onto com.google.firebase:firebase-core@{strictly 16.0.9}
-- Project 'app' depends onto com.google.android.gms:play-services@{strictly 11.2.0}
-- Project 'app' depends onto com.google.firebase:firebase-iid@{strictly 20.0.0}
-- Project 'app' depends onto com.google.firebase:firebase-analytics@{strictly 16.5.0}
-- Project 'app' depends onto com.google.firebase:firebase-common@{strictly 19.0.0}
-- Project 'app' depends onto com.google.android.gms:play-services-appinvite@{strictly 11.2.0}
-- Project 'app' depends on project 'firebase_messaging' which depends onto com.google.firebase:firebase-messaging@20.0.
0
-- Project 'app' depends onto com.google.firebase:firebase-auth@{strictly 17.0.0}
-- Project 'app' depends onto com.google.firebase:firebase-crash@{strictly 11.2.0}
-- Project 'app' depends onto com.google.android.gms:play-services-measurement-api@{strictly 16.5.0}
-- Project 'app' depends onto com.google.firebase:firebase-analytics-impl@{strictly 11.2.0}
-- Project 'app' depends onto com.google.firebase:firebase-messaging@{strictly 20.0.0}
-- Project 'app' depends on project 'firebase_auth' which depends onto com.google.firebase:firebase-auth@17.0.0
-- Project 'app' depends onto com.google.firebase:firebase-database-connection@{strictly 11.2.0}
-- Project 'app' depends onto com.google.firebase:firebase-auth-interop@{strictly 17.0.0}

For extended debugging info execute Gradle from the command line with ./gradlew --info :app:assembleDebug to see the dep
endency paths to the artifact. This error message came from the google-services Gradle plugin, report issues at https://
github.com/google/play-services-plugins and disable by adding "googleServices { disableVersionCheck = false }" to your b
uild.gradle file.

**Comment by x8Haider** [comment]
I tried to clean it and re-run the run build command I found that is failing at this line before the error are thrown:

[ +283 ms] > Task :app:mergeDebugResources
[ +899 ms] > Task :app:processDebugResources FAILED
[+1299 ms] > Task :app:javaPreCompileDebug
[        ]          *********************************************************
[        ] WARNING: This version of android_intent will break your Android build if it or its dependencies aren't compatible with AndroidX.
[        ]          See https://goo.gl/CP92wY for more information on the problem and how to fix it.
[        ]          This warning prints for all Android build failures. The real root cause of the error may be unrelated.
[        ]          *********************************************************



Any ideas?

**Comment by x8Haider** [comment]
Hello all, 

I was able to crack it... I followed the link: https://flutter.dev/docs/development/packages-and-plugins/androidx-compatibility

and downgraded Gradle from 5 to 4.10.2 and added  'com.google.firebase:firebase-auth:19.0.0' to the dependencies, synced Gradle and flutter clean & the build succeeds.


It is pretty strange as I had upgraded Gradle long time back and everything was going smoothly with no issue!

Anyway, this might not the issue but no idea exactly what happened.
